docs/Developers/Adding_CI.rst

```diff
@@ -0,0 +1,129 @@
+We recommend you to add all the CI and test everything in your
+own repository before doing a PR.
+
+Before we start adding CI it's important you have a GitHub account
+and know how to fork repositories. In case you don't, you should
+have a look into our `Git Tutorial <https://coala.io/gitbasics>`_.
+
+Travis CI
+---------
+
+Travis is used to confirm that the tools install and build
+properly. It also runs the tests and confirms all test cases
+pass and have 100% coverage. These are examples of travis CI
+checks used in coala and coala-bears repository:
+https://travis-ci.org/coala/coala/ and
+https://travis-ci.org/coala/coala-bears/.
+To run identical CI checks in travis you will need to configure your
+forked repository and to do that follow the steps mentioned below.
+
+1. Go to `travis-ci.org <https://travis-ci.org/>`_ and create
+   an account. You can simply use your GitHub account for that.
+2. On the top left corner you will see a "+" icon beside
+   "My Repositories". Click on that, and it will take you to
+   your travis-ci profile.
+3. Sync your account with github by clicking on the top right
+   button saying "Sync account".
+4. Find the forked coala repository in the list and enable builds
+   for it.
+5. Travis CI requires a .travis.yml file containing the settings
+   and build instructions, e.g `coala's .travis.yml
+   <https://github.com/coala/coala/blob/master/.travis.yml>`_.
+   Your forked repository from coala will already have that file.
+6. Watch the builds at
+   travis-ci.org/<username>/<repository>/builds.
+
+AppVeyor CI
+-----------
+
+To find out how coala acts in Microsoft Windows, we use
+AppVeyor which runs test and build commands in a
+Microsoft Windows box. Here are examples of CI build in AppVeyor :
+https://ci.appveyor.com/project/coala/coala/ and
+https://ci.appveyor.com/project/coala/coala-bears/.
+Now to add an indentical Appveyor CI to your forked repository,
+follow the following instructions.
+
+1. Go to `ci.appveyor.com <http://ci.appveyor.com>`_ and login
+   using your GitHub account.
+2. Click on "New Project" and find forked repository from the
+   repositories listed under your username.
+3. On the right side, you will see an "Add" button, click on it
+   and it will add it to your projects.
+4. AppVeyor CI requires appvyor.yml file that should have the
+   settings and instructions for windows, e.g `coala's appveyor.yml
+   <https://github.com/coala/coala/blob/master/.misc/appveyor.yml>`_.
+   Your forked repository already has that file.
+5. In case it has a different name or not in the root directory you
+   have to configure it in the settings which can be found at
+   ci.appveyor.com/project/<username>/<repository>/settings.
+   For coala's repository the appveyor.yml file is inside the .misc
+   directory. So you have to go to Settings and under
+   "Custom configuration .yml file name", enter
+   :code:`.misc/appveyor.yml`. If you have forked a different
```

Comment:
.ci/appveyor.yml for coala-bears and also specify the branch you want to build in ci.appveyor.com/project/<username>/<repository>/settings